spec/forms/providers/means/regular_income_form_spec.rb

```diff
@@ -0,0 +1,289 @@
+require "rails_helper"
+
+# rubocop:disable Rails/SaveBang
+RSpec.describe Providers::Means::RegularIncomeForm do
+  describe "validations" do
+    context "when neither a income type or none are selected" do
+      it "is invalid" do
```

Comment:
I don't feel comfortable with the complete change in style for our rspec tests. 

I think we should keep the existing pattern of using rspec `before/after/around` and `let` to extract out setup as and keep the individual example "thin", as much as possible pending a wider cross-team discussion. 

Personally, i find this style less clear and less open to DRYing out, if appropriate, than existing styles and feels like a backward step to me.

I have already added this [ticket to our trello board](https://trello.com/c/POzgiM2O/127-use-of-before-and-lets-in-specs) to start that off.